File: Leetcode/1260.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int n = (int)grid.size();
        int m = (int)grid.front().size();
        int total = n * m;
        vector<int> v(total);
        for (int i = 0, c = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                v[c++] = grid[i][j];
            }
        }
        rotate(v.rbegin(), v.rbegin() + (k % total) , v.rend());
        for (int i = 0, c = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                grid[i][j] = v[c++];
            }
        }
        return grid;
    }
};

File: Leetcode/1260/1260.go
func shiftGrid(grid [][]int, k int) [][]int {
	var temp []int
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			temp = append(temp, grid[i][j])
		}
	}
	if k > len(temp) {
		k %= len(temp)
	}
	var temp2 []int
	for i := len(temp)-k; i < len(temp); i++ {
		temp2 = append(temp2, temp[i])
	}
	for i := 0; i < len(temp)-k; i++ {
		temp2 = append(temp2, temp[i])
	}
	pos := 0
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			grid[i][j] = temp2[pos]
			pos++
		}
	}
	return grid
}